tianshou/data/buffer.py

```diff
@@ -520,13 +437,215 @@ def update_weight(
     def __getitem__(
         self, index: Union[slice, int, np.integer, np.ndarray]
     ) -> Batch:
-        return Batch(
-            obs=self.get(index, "obs"),
-            act=self.act[index],
-            rew=self.rew[index],
-            done=self.done[index],
-            obs_next=self.get(index, "obs_next"),
-            info=self.get(index, "info"),
-            policy=self.get(index, "policy"),
-            weight=self.weight[index],
-        )
+        batch = super().__getitem__(index)
+        batch.weight = self.weight[index]
+        return batch
+
+
+class ReplayBuffers(ReplayBuffer):
```

Comment:
I suggest we do not make offset a value inside each buffer(because it belongs to replaybuffers that buffer). we can use a list or dict instead?